build_extra/rust/BUILD.gn

```diff
@@ -580,3 +581,231 @@ rust_crate("tokio_threadpool") {
     ":tokio_executor",
   ]
 }
+
+rust_crate("hyper_rustls") {
+  source_root = "$registry_github/hyper-rustls-0.14.0/src/lib.rs"
+  extern = [
+    ":ct_logs",
+    ":futures",
+    ":http",
+    ":hyper",
+    ":rustls",
+    ":tokio_core",
+    ":tokio_io",
+    ":tokio_rustls",
+    ":tokio_tcp",
+    ":webpki",
+    ":webpki_roots",
+  ]
+}
+
+ring_root = "$registry_github/ring-0.13.2/"
+
+component("ring_primitives") {
+  sources = [
+    "$ring_root/crypto/constant_time_test.c",
+    "$ring_root/crypto/cpu-aarch64-linux.c",
+    "$ring_root/crypto/cpu-arm-linux.c",
+    "$ring_root/crypto/cpu-arm.c",
+    "$ring_root/crypto/cpu-intel.c",
+    "$ring_root/crypto/crypto.c",
+    "$ring_root/crypto/fipsmodule/aes/aes.c",
+    "$ring_root/crypto/fipsmodule/aes/internal.h",
+    "$ring_root/crypto/fipsmodule/bn/exponentiation.c",
+    "$ring_root/crypto/fipsmodule/bn/generic.c",
+    "$ring_root/crypto/fipsmodule/bn/internal.h",
+    "$ring_root/crypto/fipsmodule/bn/montgomery.c",
+    "$ring_root/crypto/fipsmodule/bn/montgomery_inv.c",
+    "$ring_root/crypto/fipsmodule/bn/shift.c",
+    "$ring_root/crypto/fipsmodule/cipher/e_aes.c",
+    "$ring_root/crypto/fipsmodule/cipher/internal.h",
+    "$ring_root/crypto/fipsmodule/ec",
+    "$ring_root/crypto/fipsmodule/ec/ecp_nistz.c",
+    "$ring_root/crypto/fipsmodule/ec/ecp_nistz.h",
+    "$ring_root/crypto/fipsmodule/ec/ecp_nistz256.c",
+    "$ring_root/crypto/fipsmodule/ec/ecp_nistz256.h",
+    "$ring_root/crypto/fipsmodule/ec/ecp_nistz384.h",
+    "$ring_root/crypto/fipsmodule/ec/gfp_p256.c",
+    "$ring_root/crypto/fipsmodule/ec/gfp_p384.c",
+    "$ring_root/crypto/fipsmodule/modes/gcm.c",
+    "$ring_root/crypto/fipsmodule/modes/internal.h",
+    "$ring_root/crypto/internal.h",
+    "$ring_root/crypto/limbs/limbs.c",
+    "$ring_root/crypto/limbs/limbs.h",
+    "$ring_root/crypto/mem.c",
+    "$ring_root/include/GFp/aes.h",
+    "$ring_root/include/GFp/arm_arch.h",
+    "$ring_root/include/GFp/base.h",
+    "$ring_root/include/GFp/cpu.h",
+    "$ring_root/include/GFp/mem.h",
+    "$ring_root/include/GFp/type_check.h",
+    "$ring_root/third_party/fiat/curve25519.c",
+    "$ring_root/third_party/fiat/curve25519_tables.h",
+    "$ring_root/third_party/fiat/internal.h",
+
+    #"$ring_root/crypto/fipsmodule/modes/polyval.c",
+  ]
+  if (is_mac) {
+    sources += [
+      "$ring_root/pregenerated/aes-586-macosx.S",
```

Comment:
Did you compile these or were they in the crate?

Reply:
In the crate.